tuned/exports/unix_socket_exporter.py

```diff
@@ -0,0 +1,195 @@
+		if not ismethod(method):
+			raise Exception("Only bound methods can be exported.")
+		
+		method_name = method.__name__
+		if method_name in self._unix_socket_methods:
+			raise Exception("Method with this name (%s) is already exported." % method_name)
+		
+		class wrapper(object):
+			def __init__(self, out_signature):
+				self._out_signature = out_signature
+			
+			def __call__(self, *args, **kwargs):
+				return method(*args, **kwargs)
+		
+		self._unix_socket_methods[method_name] = wrapper(out_signature)
+		self._signals.add(method_name)
+
+	def send_signal(self, signal, *args, **kwargs):
+		if not signal in self._signals:
+			raise Exception("Signal '%s' doesn't exist." % signal)
+		for p in self._socket_signal_paths:
+			try:
+				s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+				s.setblocking(False)
+				s.connect(p)
+				self._send_data(s, "sig", list(args) + list(kwargs.values()))
+				s.close()
+			except OSError as e:
+				log.warning("Error while sending signal '%s' to socket '%s': %s" % (signal, p, e))
+
+	def register_signal_path(self, path):
+		self._socket_signal_paths.append(path)
+
+	def _construct_socket_object(self):
+		if self._socket_path:
+			if os.path.exists(self._socket_path):
+				os.unlink(self._socket_path)
+			self._socket_object = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+			#self._socket_object.setblocking(False)
+			self._socket_object.bind(self._socket_path)
+			self._socket_object.listen(16)
+			os.chown(self._socket_path, self._ownership[0], self._ownership[1])
+			if self._permissions:
+				os.chmod(self._socket_path, self._permissions)
+
+	def start(self):
+		if self.running():
+			return
+
+		self.stop()
+		self._construct_socket_object()
+
+	def stop(self):
+		if self._socket_object:
+			self._socket_object.close()
+
+	def _send_data(self, s, type, data):
+		"""
+		Sends data via socket s.
+		First byte of message specifies type of message (currently "r" for response, "e" for error message and "s" for signal).
+		Rest of message is json data.
+		"""
+		log.debug("Sending socket data (type: %s, data: %s)" % (type, data))
+		try:
+			s.send(type[0].encode("utf-8") + json.dumps(data).encode("utf-8"))
+		except Exception as e:
+			log.warning("Failed send data '%s' type '%s': %s" % (data, type, e))
+
+	def period_check(self):
+		"""
+		Periodically checks socket object for new calls. This allows to function without special thread.
+		Message expects json array, where first item is name of called method, rest is arguments of given method.
+		For response see method _send_data.
+		
+		Example call:
+		
+		printf '["active_profile"]' | sudo nc -U /run/tuned/tuned.sock
+		printf '["switch_profile", "balanced"]' | sudo nc -U /run/tuned/tuned.sock
+		"""
+		r, _, _ = select.select([self._socket_object], (), (), 0)
+		if r:
+			conn, _ = self._socket_object.accept()
+			try:
+				data = ""
+				while True:
+					rec_data = conn.recv(4096).decode()
+					if not rec_data:
```

Comment:
Yeah, unfortunately, this doesn't scale.  Even this fails for some of the requests:

```
send() {
  local data="$1"
  printf "$data" | nc -U /run/tuned/tuned.sock
}

for i in $(seq 1 10)
do
  send '["active_profile"]' &
done
```

Reply:
Can add this const to config so you can set how many connections to listen to. Can add to hold connection too, but I don't think it would fix this scenario